### PR TITLE
[Many ports] Update to the latest version or commit

### PR DESCRIPTION
**Pull request by NancyLi1013**
Update ports to latest version and also fix build errors.

Update ci.baseline.txt

Related #13536

Note: There are many errors in the latest version for `librsync`. So I updated it to the latest commit.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: ports/libmaxminddb/fix-linux-build.patch [hunk 1]
```diff
@@ -1,19 +1,19 @@
-diff --git a/include/maxminddb.h b/include/maxminddb.h
```
Comment (ras0219-msft) [inline ports/libmaxminddb/fix-linux-build.patch:1]: Please restore the line endings of this file to the original type (probably LF?)

File: ports/libplist/CMakeLists.txt [hunk 2]
```diff
@@ -22,12 +22,12 @@ install(
     LIBRARY DESTINATION lib
 )
 configure_file(
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/libplist.pc.in
-    ${CMAKE_CURRENT_BINARY_DIR}/libplist.pc
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/libplist++-2.0.pc.in
```
Comment (ras0219-msft) [inline ports/libplist/CMakeLists.txt:25]: There are now two .pc files that each need to be installed.

File: ports/libplist/portfile.cmake [hunk 3]
```diff
@@ -21,4 +21,4 @@ vcpkg_copy_pdbs()
```
Comment (ras0219-msft) [inline ports/libplist/portfile.cmake:21]:
```suggestion
vcpkg_fixup_pkgconfig()

```

File: ports/librsync/portfile.cmake [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
+vcpkg_fail_port_install(ON_TARGET "uwp")
 
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO librsync/librsync
-    REF v2.0.2
-    SHA512 5d2bc1d62b37e9ed7416203615d0a0e3c05c4c884b5da63eda10dd5c985845b500331bce226e4d45676729382c85b41528282e25d491afda31ba434ac0fefad7
+    REF d1938c387e86ab5bbf7cb2e84244229c5bbd5ebf # commit 2020-06-04
```
Comment (ras0219-msft) [inline ports/librsync/portfile.cmake:8]: We should use the latest release: https://github.com/librsync/librsync/releases/tag/v2.3.1

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/librsync/portfile.cmake (hunk 4 above)
Comment (NancyLi1013) [reply ports/librsync/portfile.cmake:8]: There are some errors in the latest release, so I used the latest commit to update this port.
Such as 
```
CMake Error at C:/Program Files/CMake/share/cmake-3.17/Modules/TestBigEndian.cmake:50 (message):
  no suitable type found
Call Stack (most recent call first):
  CMakeLists.txt:115 (TEST_BIG_ENDIAN)
```
Please see https://github.com/librsync/librsync/issues/207

`error C2491: 'rs_trace_to': definition of dllimport function not allowed` 

These two issues have been fixed in the latest commit.

File: ports/libmaxminddb/fix-linux-build.patch (hunk 1 above)
Comment (NancyLi1013) [reply ports/libmaxminddb/fix-linux-build.patch:1]: Thanks for your reminder. I will update this later.

File: ports/libplist/CMakeLists.txt (hunk 2 above)
Comment (NancyLi1013) [reply ports/libplist/CMakeLists.txt:25]: Thanks for your review. I ignored this point before. Will update it.

**Comment by NancyLi1013** [comment]
pkgconfig for `libplst `needs to further fix. So I convert this PR to draft until the issue is fixed.

**Comment by NancyLi1013** [comment]
Revert `libplist` and will update it in another PR.

**Review by JackBoosY — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**